components/dialog/dialog.js

```diff
@@ -164,6 +193,10 @@ class Dialog extends LocalizeCoreElement(AsyncContainerMixin(DialogMixin(LitElem
 		this._hasFooterContent = (footerContent && footerContent.length > 0);
 	}
 
+	_handleResize() {
+		this._autoSize = !mediaQueryList.matches;
```

Comment:
We need to turn off the autosize attribute to get the dialog to show up as a "fullscreen" in mobile mode. Otherwise, it will just fit the content and not extend to the bottom of the screen as required. The code for the mediaQueryList and listener is the same as d2l-dialog-fullscreen. I also made sure to keep the check for the addEventListener function to avoid that Safari 13 issue with compatibility.